### PR TITLE
Refactory katello_repository to utilize cement

### PR DESCRIPTION
**Pull request by gregswift**
Includes the fixed tests from mdellweg (but I did a lazy pull so lost the commit)

Pulls changes from #79

### CODE REVIEW

File: module_utils/ansible_nailgun_cement.py [hunk 1]
```diff
@@ -224,6 +224,11 @@ def delete_entity(entity, module):
     return True
 
 
+def find_content_views(module, content_views, organization, failsafe=False):
+    return map(lambda content_view: find_content_view(module, content_view['name'], content_view['version'],
+                                                      organization, failsafe=failsafe), content_views)
+
+
```
Comment (mdellweg) [inline module_utils/ansible_nailgun_cement.py]: I think, this is unrelated. It should be in some 'refactor_content_view' PR.

File: test/test_playbooks/tasks/repository.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
+- name: "Create katello repository"
+  vars:
+    - organization_name: "Test Organization"
+    - product_name: "Test Product"
+    - repository_name: "Test Repository"
+    - repository_content_type: "yum"
+    - repository_state: present
```
Comment (mdellweg) [inline test/test_playbooks/tasks/repository.yml]: Same as in #95 ,
so maybe wait for the decision there.

File: modules/katello_repository.py [hunk 3]
```diff
@@ -73,6 +73,13 @@
             - immediate
             - on_demand
         required: false
+    state:
+        description:
+            - State of the Repository
+        default: true
+        choices:
+        - present
+        - absent
```
Comment (mdellweg) [inline modules/katello_repository.py]: Do you mind adding 'present_with_defaults' here and inline 148?
Cements naildown should make it _just work_.

File: module_utils/ansible_nailgun_cement.py (hunk 1 above)
Comment (rackergs) [reply module_utils/ansible_nailgun_cement.py]: doh.. i meant to leave that out. thanks

**Comment by mdellweg** [comment]
One last (hopefully) thing i just saw: Can you please add the state parameter in the example string.
I think, then we are really good to go.
